include/swift/AST/DiagnosticConsumer.h

```diff
@@ -209,16 +286,15 @@ class FileSpecificDiagnosticConsumer : public DiagnosticConsumer {
   /// Xcode will only see an error for a particular primary in that primary's
   /// serialized diagnostics file. So, tell the subconsumers to inform the
   /// driver of incomplete batch mode compilation.
-  void tellSubconsumersToInformDriverOfIncompleteBatchModeCompilation() const;
+  void tellSubconsumersToInformDriverOfIncompleteBatchModeCompilation();
 
   void computeConsumersOrderedByRange(SourceManager &SM);
 
   /// Returns nullptr if diagnostic is to be suppressed,
   /// None if diagnostic is to be distributed to every consumer,
   /// a particular consumer if diagnostic goes there.
-  Optional<ConsumerSpecificInformation *>
-  consumerSpecificInformationForLocation(SourceManager &SM,
-                                         SourceLoc loc) const;
+  Optional<FileSpecificDiagnosticConsumer::Subconsumer *>
```

Comment:
Nitpick: no need to qualify with FileSpecificDiagnosticConsumer within the implementation.